---
title: "Pa Compendia Quality Control: application"
output: html_notebook
author: Georgia Doing
date: Sept 23, 2021
---

```{r, include=FALSE}
knitr::opts_chunk$set(
  echo = FALSE, warning = F, message=F
)
```

#### Background

In the genesis of this compendium of RNAseq data, a wide net was cast on the NCBI SRA database in order to maximize the number of publicly available P. aeruginosa RNAseq datasets collected. For details on the search project see [documentation](https://docs.google.com/document/d/1OnvJNVkhK5ATnhHaeSmz8Ze_Iw9ixXhfSxAL_BSqbuM/edit).

In order to ensure the quality of the compendium, we need to impose some heuristic and data-driven inclusion criteria. To do so we will look at some high-level characteristics of the compendium before narrowing in the the specific criteria we will use, acknowledging that they are specific to this compendium and may not necessarily generalize to other compendia.

#### Outline

1. [Setup and data import](## 1. Setup and data import)
2. [Strain and HK gene annotations](# 2. Load Annotations)
3. [Set criteria](## 3. Set Final Criteria)
4. [Apply criteria](## 4. Apply)
5. [Normalization](# 5. Normalization)
6. [Save filtered compendia](## 6. Write out Data)
7. [Summary of datasets](# 7. Summary of filtered datasets)

## 1. Setup and data import


```{r}
library(kableExtra)
source('annotation_functions.R')
source('filter_functions.R')
source('plotting_functions.R')
```


Load the PAO1- and PA14- aligned RNAseq compendia in both estimated counts and TPM forms resulting from the Salmon workflow.


Note that row names are converted from the ensemble cDNA gene IDs to PAO1 or PA14 numbers.

```{r}
# PAO1
rnaseq_pao1_tpm <- read.csv('compendia_download/TPM_pao1_cdna_k15.csv', 
                            stringsAsFactors = F, row.names = 1)[,-1]
rownames(rnaseq_pao1_tpm) <- make.unique(sapply(rownames(rnaseq_pao1_tpm), 
                                                function(x) cDNA_to_PAO1(x)))
rnaseq_pao1_counts <- read.csv('compendia_download/num_reads_pao1_cdna_k15.csv', 
                               stringsAsFactors = F, row.names = 1)[,-1]
rownames(rnaseq_pao1_counts) <- make.unique(sapply(rownames(rnaseq_pao1_counts), 
                                                   function(x) cDNA_to_PAO1(x)))

# PA14
rnaseq_pa14_tpm <- read.csv('compendia_download/TPM_pa14_cdna_k15.csv', 
                            stringsAsFactors = F, row.names = 1)[,-1]
rownames(rnaseq_pa14_tpm) <- make.unique(sapply(rownames(rnaseq_pa14_tpm), 
                                                function(x) cDNA_to_PA14(x)))
rnaseq_pa14_counts <- read.csv('compendia_download/num_reads_pa14_cdna_k15.csv', 
                               stringsAsFactors = F, row.names = 1)[,-1]
rownames(rnaseq_pa14_counts) <- make.unique(sapply(rownames(rnaseq_pa14_counts), 
                                                   function(x) cDNA_to_PA14(x)))
```

For filtering applications I use log-transformed data since it resembles more normal distributions. the hlog ("happy log") function simply takes the log of a dataframe making sure the resulting data frame doesn't have any -Inf or NaN values.

```{r}
rnaseq_pao1_tpm_log <- hlog(rnaseq_pao1_tpm)
rnaseq_pa14_tpm_log <- hlog(rnaseq_pa14_tpm)
```

# 2. Load Annotations

We do not have comprehensive annotations at this point but can use the SRA run table to gather as much data about experiments in our compendium as is easily possible.

```{r}
run_table <- read.csv('compendia_download/SraRunTable.csv', 
                      stringsAsFactors = F)
```

Since the compendium is composed of multiple strains and filtering criteria are based on characteristic distributions across the compendium, we want to be able to build those distributions from samples aligned to their matched target reference to avoid alignment-based artifacts influencing the distributions. To do this we will scrape all fields of the SRA run table to look for "PAO1" and "PA14" to provide best-guesses of what strain each sample is, if annotated.

```{r}
# search for PAO1
seq_strain_ann_pao1 <- rowSums(apply(run_table, 2, FUN = function(x){
  sapply(x, function(y){
    grepl('PAO1',y) | grepl('PA01',y) # since PAO1 is sometimes misspelled as PA01
  })
})) > 0
# search for PA14
seq_strain_ann_pa14 <- rowSums(apply(run_table, 2, FUN = function(x){
  sapply(x, function(y){
    grepl('PA14',y)
  })
})) > 0
# consolidate PAO11 and PA14 annotations
seq_strain_ann <- sapply(c(1:nrow(run_table)), function(x){
  if(seq_strain_ann_pao1[x]){
    'PAO1'
  } else if(seq_strain_ann_pa14[x]){
    'PA14'
  } else{'No annotation'}
})
# name the strain annotations to match compendium for easy indexing
names(seq_strain_ann) <- paste(run_table$Experiment,
                               run_table$Experiment,'salmon',sep='.')
# re-order strain annotations  to path compendium samples
seq_strain_ann_order <- seq_strain_ann[colnames(rnaseq_pao1_tpm)]
```

Using genes known to be stable in their expression as a normalization metric is commonly used in qRT analysis and other mRNA quantification techniques. It can provide internal controls and account for technical biases. The same concept can hold true when we look at gene expression across compendia. We can choose housekeeping genes for **P. aeruginosa** based on a publication and examine their expression in the array compendium.

The following set of HK genes comes from Alqarni et al 2016, J Microbiol Methods. 

```{r}
## PAO1
hks_pao1 <- sapply(c('ppiD','rpoD','rpoS','proC','recA','rpsL','rho','oprL',
                     'tipA','nadB'), 
              function(x) name_to_PAO1(x))
# only include hk genes if they are in our compendium data
hks_pao1 <- hks_pao1[hks_pao1 %in% rownames(rnaseq_pao1_tpm_log)]
## PA14
hks_pa14 <- sapply(c('ppiD','rpoD','rpoS','proC','recA','rpsL','rho','oprL',
                     'tipA','nadB'), 
              function(x) name_to_PA14(x))
hks_pa14 <- hks_pa14[hks_pa14 %in% rownames(rnaseq_pa14_tpm_log)]
```


## 3. Set Final Criteria

We will use the distributions of zero counts and median hk gene expression across the compendium to determine lower and upper thresholding criteria to exclude outlying samples. Using the PAO1- and PA14-aligned data, we can tailor our thresholds to avoid misalignment artifacts by basing the thresholds for each strain on samples of that matched strain.

```{r}
# zero counts lower and upper thresholds
pao1_zeros <- get_zeros(
  rnaseq_pao1_tpm_log[,seq_strain_ann_order == 'PAO1'], .1, .9)
pa14_zeros <- get_zeros(
  rnaseq_pa14_tpm_log[,seq_strain_ann_order == 'PA14'], .1, .9)
# median hk expression lower and upper thresholds
pao1_hk <- get_hks(
  rnaseq_pao1_tpm_log[,seq_strain_ann_order == 'PAO1'], 
  hks_pao1, .2, .98)
pa14_hk <- get_hks(
  rnaseq_pa14_tpm_log[,seq_strain_ann_order == 'PA14'], 
  hks_pa14, .2,.98)
```


## 4. Apply

Now that upper and lower thresholds have been determined, we can apply these criteria across the compendium. I apply the zero-based and hk gene-based as well as the PAO1-based and PA14-based criteria separately and then consolidate in order to be able to re-trace which tests each samples passed and failed.

```{r}
## PAO1
filt_sp_pao1 <- filter_sparsity(rnaseq_pao1_tpm_log,
                                max_zeros=pao1_zeros[2],
                                min_zeros=pao1_zeros[1])
filt_hk_pao1 <- filter_hks(rnaseq_pao1_tpm_log, hks_pao1,
                           hk_min=pao1_hk[1],
                           hk_max=pao1_hk[2])
filt_pao1_samp_i <- (filt_sp_pao1 & filt_hk_pao1)
## PA14
filt_sp_pa14 <- filter_sparsity(rnaseq_pa14_tpm_log,
                                max_zeros=pao1_zeros[2],
                                min_zeros=pa14_zeros[1])
filt_hk_pa14 <- filter_hks(rnaseq_pa14_tpm_log, hks_pa14,
                           hk_min=pa14_hk[1],
                           hk_max=pa14_hk[2])
filt_pa14_samp_i <- (filt_sp_comp_pa14_tpm & filt_hk_comp_pa14_tpm)
# consolidate to include samples that pass PAO1 or PA14 based criteria
filt_i <- filt_pao1_samp_i | filt_pa14_samp_i
```

For the final compendia we will keep samples that passed either both PAO1 zero- and hk-based criteria or both PA14 zero- and hk-based criteria. In this way the PAO1 and PA14 comepndia will contain the exact same set of samples, just aligned to difference references.

```{r}
## PAO1
out_rnaseq_pao1_tpm <- rnaseq_pao1_tpm[rownames(rnaseq_pao1_tpm), filt_i]
out_rnaseq_pao1_counts <- rnaseq_pao1_counts[rownames(rnaseq_pao1_tpm), filt_i]
## PA14
out_rnaseq_pa14_tpm <- rnaseq_pa14_tpm[rownames(rnaseq_pa14_tpm), filt_i]
out_rnaseq_pa14_counts <- rnaseq_pa14_counts[rownames(rnaseq_pa14_tpm), filt_i]
```



# 5. Normalization

After filtering out outlier datasets there is still a wide range of gene expression distributions that vary sample-to-sample. This is in part due to differences in read depth, though not entirely corrected by read depth normalization such as TPM. This problem has been tackled for differential expression analysis and so we employed a normalization methods used in the DESeq2 method of DE analysis. This method, referred to as the median ratios method (MR) calculates per sample size factors based on the ratio of median expression in each sample to the median expression of a pseudo-reference sample defined as the geometric mean of all samples in the compendium. 

```{r}
normalized_counts_pao1 <- MRnorm(out_rnaseq_pao1_counts)
normalized_counts_pa14 <- MRnorm(out_rnaseq_pa14_counts)
```


## 6. Write out Data

And now, at long last, we shall save these compendia in TPM, counts snf normalized counts form noting the 'p2' in the title indicating the parameter combination used for filtering. 
```{r}
#### PAO1
write.csv(out_rnaseq_pao1_tpm, 
          'qc-out/pao1_aligned_compendium_p2_filtered_tpm.csv')

write.csv(out_rnaseq_pao1_counts, 
          'qc-out/pao1_aligned_compendium_p2_filtered_counts.csv')

write.csv(normalized_counts_pao1, 
          'qc-out/pao1_aligned_compendium_p2_filtered_counts_norm.csv')

#### PA14
write.csv(out_rnaseq_pa14_tpm, 
          'qc-out/pa14_aligned_compendium_p2_filtered_tpm.csv')

write.csv(out_rnaseq_pa14_counts, 
          'qc-out/pa14_aligned__compendium_p2_filtered_counts.csv')

write.csv(normalized_counts_pa14, 
          'qc-out/pa14_aligned_compendium_p2_filtered_counts_norm.csv')
```



# 7. Summary of filtered datasets

For a quick summary of the filtering process we can visualize the distributions of zero counts and median hk gene expression for the PAO1- and PA14-aligned compendia for their respectively strain-matched samples and for all samples. We can also check out a couple tables showing the experiments with the highest number of samples that remained after filtering and the experiments with the highest number of samples that were removed.

```{r, warning=F}
#pdf('zeros_dists.pdf', width = 6, height=3)
par(mfrow=c(2,2))
pao1_seq_zeros <- apply(rnaseq_pao1_counts, 2, FUN = function(x) sum( x  == 0))
pa14_seq_zeros <- apply(rnaseq_pa14_counts, 2, FUN = function(x) sum( x  == 0))

hist(log(pao1_seq_zeros[seq_strain_ann_order == 'PAO1']), 
     main = 'PAO1\n PAO1 samples', 
     breaks=25, xlab = 'Zeros')
abline(v = log(pao1_zeros[1]), col='red', lty=2)
abline(v = log(pao1_zeros[2]), col='red', lty=2)

hist(log(pa14_seq_zeros[seq_strain_ann_order == 'PA14']), 
     main = 'PA14\n PA14 samples', 
     breaks=25,xlab = 'Zeros')
abline(v = log(pa14_zeros[1]), col='red', lty=2)
abline(v = log(pa14_zeros[2]), col='red', lty=2)

hist(log(pao1_seq_zeros[seq_strain_ann_order != 'PAO1']), main = 'PAO1\n all samples', 
     breaks=25, xlab = 'Zeros')
abline(v = log(pao1_zeros[1]), col='red', lty=2)
abline(v = log(pao1_zeros[2]), col='red', lty=2)

hist(log(pa14_seq_zeros[seq_strain_ann_order != 'PA14']), main = 'PA14\n all samples', 
     breaks=25, xlab = 'Zeros')
abline(v = log(pa14_zeros[1]), col='red', lty=2)
abline(v = log(pa14_zeros[2]), col='red', lty=2)
#dev.off()
```


Figure 1. Distributions of genes with zero counts across PAO1- and PA14-aligned compendia.

The samples in the compendium have different numbers of genes with zero counts, contributing the the different degrees of sparsity in RNA-seq data. Specifically, for PAO1 annotaed samples aligned to a PAO1 reference (top left) the number of genes with zero counts per sample ranges from `r min(pao1_seq_zeros[seq_strain_ann_order == 'PAO1'])` to `r max(pao1_seq_zeros[seq_strain_ann_order == 'PAO1'])`, with an median of `r median(pao1_seq_zeros[seq_strain_ann_order == 'PAO1'])` whereas samples not annotated as PAO1 mapped to a PAO1 reference (bottom left) have a similar range, with a higher median `r median(pao1_seq_zeros[seq_strain_ann_order != 'PAO1'])`, with more genes with zero counts than samples annotated as PAO1. The same trend holds for samples annotated as PA14 mapped to PA14 (top right) with a median of `r median(pa14_seq_zeros[seq_strain_ann_order == 'PA14'])` and samples not annotated as PA14 mapped to PA14 (bottom right) with a median of `r median(pa14_seq_zeros[seq_strain_ann_order != 'PA14'])`.




```{r, warning=F}
#pdf("hk_genes_pao1_pa14_w_cutoffs.pdf", width=6,height=6)
par(mfrow=c(2,2))
hk_means_pao1 <- apply(rnaseq_pao1_tpm_log[hks_pao1,], 2, 
                       FUN = function(x) mean(x))
hk_means_pa14 <- apply(rnaseq_pa14_tpm_log[hks_pa14,], 2, 
                       FUN = function(x) mean(x))
hist(hk_means_pao1[seq_strain_ann_order == 'PAO1'], 
     main = 'PAO1-aligned\n PAO1 samples', xlab = 'median HK expression')
abline(v = pao1_hk[1], col='red', lty=2)
abline(v = pao1_hk[2], col='red', lty=2)

hist(hk_means_pa14[seq_strain_ann_order == 'PA14'], 
     main = 'PA14-aligned\n PA14 samples', xlab = 'median HK expression')
abline(v = pa14_hk[1], col='red', lty=2)
abline(v = pa14_hk[2], col='red', lty=2)

hist(hk_means_pao1[seq_strain_ann_order != 'PAO1'], 
     main = 'PAO1-aligned\n all samples', xlab = 'median HK expression')
abline(v = pao1_hk[1], col='red', lty=2)
abline(v = pao1_hk[2], col='red', lty=2)

hist(hk_means_pa14[seq_strain_ann_order != 'PA14'], 
     main = 'PA14-aligned\n all samples', xlab = 'median HK expression')
abline(v = pa14_hk[1], col='red', lty=2)
abline(v = pa14_hk[2], col='red', lty=2)
#dev.off()
```

Figure 2. Distributions of median housekeeping gene expression across PAO1- and PA14-aligned compendia.

Unlike the distributions of zero count genes, the median expression of housekeeping genes is distributed simialrly across the PAO1-aligned samples annotated as PAO1 (top left) and not annotated as PAO1 (bottom left) as well as the PA14-aligned samples annotated as PA14 (top right) and not annotated as PA14 (bottom right). This is perhaps due to the conserved nature of most houskeeping genes and the easy cross mapping of strain and reference. 


We can save these statistics so the compendium can be easily filtered at different thesholds for different uses. 
```{r}
co_df <- data.frame('PAO1_zeros_count' = pao1_seq_zeros,
                    'PA14_zeros_count' = pa14_seq_zeros,
                    'PAO1_hk_median' = hk_means_pao1 ,
                    'PA14_hk_median' = hk_means_pa14 ,
                    'strain' = seq_strain_order,
                    'Experiment' = names(hk_means_pao1))
#write.csv(co_df,'cutoff_stats.csv')
```



Before we go, let's take a quick look at the experiments that make up the compendium after filtering and the experiments that were filtered out.

```{r}

# must meet all PAO1-based  or all PA14-based criteria
pao1_t <- table(ann_exp_groups[
                                 (filt_sp_comp_pao1_tpm 
                                  & filt_hk_comp_pao1_tpm) |
                                   (filt_sp_comp_pa14_tpm 
                                  & filt_hk_comp_pa14_tpm)])

pao1_only <- table(ann_exp_groups[
                                 (filt_sp_comp_pao1_tpm 
                                  & filt_hk_comp_pao1_tpm) &
                                   !(filt_sp_comp_pa14_tpm 
                                  & filt_hk_comp_pa14_tpm) ])
pa14_only <- table(ann_exp_groups[
                                 (filt_sp_comp_pa14_tpm 
                                  & filt_hk_comp_pa14_tpm) &
                                   !(filt_sp_comp_pao1_tpm 
                                  & filt_hk_comp_pao1_tpm)])

pao1_both <- table(ann_exp_groups[
                                 (filt_sp_comp_pao1_tpm 
                                  & filt_hk_comp_pao1_tpm) &
                                   (filt_sp_comp_pa14_tpm 
                                  & filt_hk_comp_pa14_tpm)])



all_t <- table(ann_exp_groups)

ugs <- unique(names(all_t))

summ_t <- data.frame('Experiment' = ugs, 
                     'passed_PAO1' = as.numeric(pao1_only[ugs]),
                     'passed_PA14' = as.numeric(pa14_only[ugs]),
                     'passed_crit' = as.numeric(pao1_t[ugs]))


ann_exp_groups[is.na(ann_exp_groups)] <- "No group"
all_t_s <- data.frame('groups' = ann_exp_groups,
                          'strain' = seq_strain_order)
rownames(all_t_s) <- make.unique(as.character(all_t_s$groups))

pao1_t <- pao1_t[ugs]
all_t <- all_t[ugs]
pao1_t[is.na(pao1_t)] <- 0
all_t[is.na(all_t)] <- 0

summ_t_strain <- data.frame('Experiment' = ugs,
                            'Strain' = all_t_s[ugs,'strain'], 
                             'passed PAO1' = as.numeric(pao1_only[ugs]),
                            'passed PA14' = as.numeric(pa14_only[ugs]),
                            'passed both' = as.numeric(pao1_both[ugs]),
                            'total passed' = as.numeric(pao1_t),
                            'failed' = as.numeric(all_t) - as.numeric(pao1_t))

summ_t_strain[is.na(summ_t_strain)] <- 0

```

```{r}
kable_classic(kable(summ_t_strain[head(order(summ_t_strain[,6], decreasing = T), 
                                       n=15),], 
                    row.names = F,
                    caption = "Top 15 experiments containing the most samples in the final, filtered compendium"))
```
A large portion of the compendium, `r   414*100 / ncol(out_rnaseq_pao1_tpm) ` % comes from a large collection of 414 clinical islates [SRP173226](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA509195), I beleive grown in LB. Two other large experiments conatinsing 190 [SRP214490](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA554358) and 122 samples ( or `r   190*100 / ncol(out_rnaseq_pao1_tpm) `) % and `r   122*100 / ncol(out_rnaseq_pao1_tpm) ` %), are also included. Although not noted as such by our temporary annotations, the 190 samples also come from a collection of clinical isolates. I believe the 122 samples come from a knock-out library of TF in PA14 [SRP034661](https://www.ncbi.nlm.nih.gov//bioproject/PRJNA231236).

```{r}
kable_classic(kable(summ_t_strain[head(order(summ_t_strain[,7], decreasing = T), 
                                       n=15),], 
                    row.names = F,
                    caption = "Top 15 experiments containing the most samples filtered out of the compendium"))
```

Of the samples that gpt filtered out, some were entire experiments such as a single-cell experiment [SRP161626](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE119888) or P. aeruginosa co-cultured with A. fumigatus [SRP168376](https://metadataplus.biothings.io/geo/GSE122391), which suggest the techinical specifics of the experimentla design may be somewhat unique compared to the bulk of rnas-seq experiments in the compendium. In other cases, such as 17 of the `r 17+122` transcription factor mutants did not pass [SRP034661](https://www.ncbi.nlm.nih.gov//bioproject/PRJNA231236) which may promote more investigation into our cut-off threshold since it is splitting up samples from the same experiment which likely were processed together.
