---
title: "Pa Compendia Quality Control: development"
output: html_notebook
author: Georgia Doing
date: June 21, 2021
---
#### Background

In the genesis of this compendium of RNAseq data, a wide net was cast on the NCBI SRA database in order to maximize the number of publicly available P. aeruginosa RNAseq datasets collected. For details on the search project see documentation: https://docs.google.com/document/d/1OnvJNVkhK5ATnhHaeSmz8Ze_Iw9ixXhfSxAL_BSqbuM/edit

In order to ensure the quality of the compendium, we need to impose some heuristic and data-driven inclusion criteria. To do so we will look at some high-level characteristics of the compendium before narrowing in the the specific criteria we will use, acknowleding that they are specific to this compendium and may not necessarily generalize to other compendia.

#### Outline

1. Setup and data import
2. Compendia integration
3. Sparsity
4. Genes with expected patterns
5. Median expression
6. Final criteria
7. Save filtered compendia

## 1. Setup and data import

### libraries and functions

```{r, message=F, warning=F}
library(ggplot2)
library(ComplexHeatmap)
library(kableExtra)
library(eulerr)
source('~/Dropbox (Hogan Lab)/Resources/Annotations/annotation_functions.R')
source('~/Dropbox (Hogan Lab)/Resources/Scripts/fsqn.R')
source('filter_functions.R')
```


### Load data

Load the array compendiu on which eADAGE was trained for comparisons and reference as well as the PAO1- and PA14- aligned RNAseq compendia in both estimated counts and TPM forms resulting from the Salmon workflow.

#### Array compendium
```{r}
array_comp <- read.csv('compendia/Pa_compendium_02_22_2014.csv', stringsAsFactors = F, sep='\t', row.names = 1)
```

#### PAO1-aligned RNAseq
Note that row names are converted from the ensemble cDNA gene IDs to PAO1 numbers.

```{r}
rnaseq_pao1_tpm <- read.csv('compendia_download/TPM_pao1_cdna_k15.csv', stringsAsFactors = F, row.names = 1)
rnaseq_pao1_tpm <- rnaseq_pao1_tpm[,-1]
rownames(rnaseq_pao1_tpm) <- make.unique(sapply(rownames(rnaseq_pao1_tpm), function(x) cDNA_to_PAO1(x)))

rnaseq_pao1_counts <- read.csv('compendia_download/num_reads_pao1_cdna_k15.csv', stringsAsFactors = F, row.names = 1)
rnaseq_pao1_counts <- rnaseq_pao1_counts[,-1]
rownames(rnaseq_pao1_counts) <- make.unique(sapply(rownames(rnaseq_pao1_counts), function(x) cDNA_to_PAO1(x)))
```

#### PA14-aligned RNAseq
Note that row names are converted from cDNA gene ID to PA14 gene numbers and then to PAO1 homologs.

```{r}
#running this chunk can take a  few minutes due to the gene ID conversions
rnaseq_pa14_tpm <- read.csv('compendia_download/TPM_pa14_cdna_k15.csv', stringsAsFactors = F, row.names = 1)
rnaseq_pa14_tpm <- rnaseq_pa14_tpm[,-1]
rownames(rnaseq_pa14_tpm) <- make.unique(sapply(rownames(rnaseq_pa14_tpm), function(x) PA14_to_PAO1(cDNA_to_PA14(x))))

rnaseq_pa14_counts <- read.csv('compendia_download/num_reads_pa14_cdna_k15.csv', stringsAsFactors = F, row.names = 1)
rnaseq_pa14_counts <- rnaseq_pa14_counts[,-1]
rownames(rnaseq_pa14_counts) <- make.unique(sapply(rownames(rnaseq_pa14_counts), function(x) PA14_to_PAO1(cDNA_to_PA14(x))))
```

## 2. Compendia integration

This is a useful step for visualizing the array and rnaseq compendia together since it matches their distributions when they otherwise have distributions with very different ranges (orders of magnitude) and shapes (somewhat normal and very negative binomial). These differences are true of both TPM and estimated counts.


Combine Array and RNAseq data using array_match(), a function defined in filter_functions.R

Note that this function returns a joint compendia with rows of genes present as row names in both input compendia (the PA14 compendia is already has row names that are PAO1 homologs) and values from the array compendium gene-specific quantile normalized (FSQN) to the RNAseq compendium.

```{r}
comp_pao1_tpm <- array_match(array_comp, rnaseq_pao1_tpm, a_ref=F)
comp_pao1_counts <- array_match(array_comp, rnaseq_pao1_counts, a_ref=F)

comp_pa14_tpm <- array_match(array_comp, rnaseq_pa14_tpm, a_ref=F)
comp_pa14_counts <- array_match(array_comp, rnaseq_pa14_counts, a_ref=F)
```







## 3. Sparsity 
Although RNAseq data can be inherently sparse, perhaps reflecting the biological nature of gene expression and the prevalence of condition-dependent or accessory genes, if a gene expression profile is too sparse it could be an indication of low read depth or low-quality reads. 



### Genes absent in too many experiments

Before assessing samples for their sparsity, I will first check for genes that may be problematic. The salmon pipeline uses the cDNA reference of PAO1 and PA14 from ensemble bacteria, which is a different reference than the genomic sequences from www.pseudomonas.com that I have used for CLC-based alignment and certainly a different collection of genes than the probes included on the **P. aeruginosa** micro-arrays.



```{r}
par(mfrow=c(1,3))
array_zeros <- apply(array_comp, 1, FUN = function(x) sum( x  < 5)) 
seqo1_zeros <- apply(rnaseq_pao1_tpm, 1, FUN = function(x) sum( x  == 0))
seq14_zeros <- apply(rnaseq_pa14_tpm, 1, FUN = function(x) sum( x  == 0))
hist(array_zeros, main = 'Low Expression (<5) per gene \n array compendium', xlab = paste('# of samples with 0 counts \n total samples: ', ncol(array_comp)), ylab = '# of genes')
abline(v=ncol(array_comp)*.75, col='red', lty=2)
hist(seqo1_zeros, main = 'Zeros per gene \n rnaseq pao1 compendium', xlab = paste('# of samples with 0 counts \n total samples: ', ncol(rnaseq_pao1_tpm)), ylab = '# of genes')
abline(v=ncol(rnaseq_pao1_tpm)*.75, col='red', lty=2)
hist(seq14_zeros, main = 'Zeros per gene \n rnaseq pa14 compendium', xlab = paste('# of samples with 0 counts \n total samples: ', ncol(rnaseq_pao1_tpm)), ylab = '# of genes')
abline(v=ncol(rnaseq_pao1_tpm)*.75, col='red', lty=2)
```
These results suggest that most genes only have 0 counts in a small number of samples, but a small handful of genes have 0 counts in most samples and are apparently only expressed in a small number of samples. This would be constant with the breadth of the compendium conditions being sufficiently large to include conditions that are conducive to the expression for most genes. 


Are the sparse genes in the array and rnaseq compendia the same genes?

```{r}

sparse_array_genes <- rownames(array_comp)
sparse_seq_genes <- rownames(rnaseq_pao1_counts)

plot(euler(c(Array=length(setdiff(sparse_array_genes,sparse_seq_genes)),
             RNAseq=length(setdiff(sparse_seq_genes,sparse_array_genes)),
                 "Array&RNAseq"=length(intersect(sparse_array_genes,sparse_seq_genes)))),
     quantities=T,
     main= 'Genes with 0 counts in at least 50% of compendium samples')



sparse_seqo1_genes <- rownames(rnaseq_pao1_counts)
sparse_seq14_genes <- rownames(rnaseq_pa14_counts)

sparse_genes_df <- data.frame('Array' = unique(c(sparse_array_genes,sparse_seqo1_genes,sparse_seq14_genes)) %in% sparse_array_genes,
                              'RNAseq_PAO1' = unique(c(sparse_array_genes,sparse_seqo1_genes,sparse_seq14_genes)) %in% sparse_seqo1_genes,
                              'RNAseq_PA14' = unique(c(sparse_array_genes,sparse_seqo1_genes,sparse_seq14_genes)) %in% sparse_seq14_genes
)
                              
plot(euler(sparse_genes_df),
     quantities=T,
     main= 'Genes with 0 counts in at least 50% of compendium samples')
plot(venn(sparse_genes_df),
     quantities=T,
     main= 'Genes with 0 counts in at least 50% of compendium samples')
```


```{r}

sparse_array_genes <- rownames(comp_pao1_tpm)[array_zeros > ncol(array_comp)*.5]
sparse_seq_genes <- rownames(comp_pao1_tpm)[seqo1_zeros > ncol(rnaseq_pao1_tpm)*.5]

plot(euler(c(Array=length(setdiff(sparse_array_genes,sparse_seq_genes)),
             RNAseq=length(setdiff(sparse_seq_genes,sparse_array_genes)),
                 "Array&RNAseq"=length(intersect(sparse_array_genes,sparse_seq_genes)))),
     quantities=T,
     main= 'Genes with 0 counts in at least 50% of compendium samples')

seqo1_zeros <- apply(rnaseq_pao1_tpm, 1, FUN = function(x) sum( x  == 0))
seq14_zeros <- apply(rnaseq_pa14_tpm, 1, FUN = function(x) sum( x  == 0))


sparse_seqo1_genes <- rownames(comp_pao1_tpm)[seqo1_zeros > ncol(rnaseq_pao1_tpm)*.5]
sparse_seq14_genes <- rownames(comp_pa14_tpm)[seq14_zeros > ncol(rnaseq_pa14_tpm)*.5]

sparse_genes_df <- data.frame('Array' = unique(c(sparse_array_genes,sparse_seqo1_genes,sparse_seq14_genes)) %in% sparse_array_genes,
                              'RNAseq_PAO1' = unique(c(sparse_array_genes,sparse_seqo1_genes,sparse_seq14_genes)) %in% sparse_seqo1_genes,
                              'RNAseq_PA14' = unique(c(sparse_array_genes,sparse_seqo1_genes,sparse_seq14_genes)) %in% sparse_seq14_genes
)
                              
plot(euler(sparse_genes_df),
     quantities=T,
     main= 'Genes with 0 counts in at least 50% of compendium samples')
```
The disjoint nature of the sets of rarely expressed genes depending on the alignment (array, PAO1 or PA14), may reflect accessory genes, though a 50% cutoff is not entire fair considering we the strain breakdown is unlikely to be 50/50 PAO1/PA14. 

The differences between the array probes and the PAO1-aligned RNAseq may be genes where the probe sequence is different than this version of the PAO1 reference that I am using. 


Check on normalization:

Do these results hold true to the array compendium after quantile normalization to RNAseq (PAO1-aligned) distributions?
```{r}
par(mfrow=c(1,2))
array_zeros <- apply(comp_pao1_tpm[,c(1:ncol(array_comp))], 1, FUN = function(x) sum( x  == 0))
seq_zeros <- apply(comp_pao1_tpm[,c(ncol(array_comp):ncol(comp_pao1_tpm))], 1, FUN = function(x) sum( x  == 0))
hist(array_zeros, main = 'Zeros per gene \n array compendium', xlab = paste('# of samples with 0 counts  \n total samples: ', ncol(array_comp)), ylab = '# of genes')
abline(v=ncol(array_comp)*.75, col='red', lty=2)
hist(seq_zeros, main = 'Zeros per gene \n rnaseq compendium', xlab = paste('# of samples with 0 counts  \n total samples: ', ncol(rnaseq_pao1_tpm)), ylab = '# of genes')
abline(v=ncol(rnaseq_pao1_tpm)*.75, col='red', lty=2)
```

All genes have expression in at least 75% of samples (noted with red dotted line) in both the array and rnaseq compendia. **I will note that since I am only looking at the combined compendia, I am only looking at the set of common genes.**




This just confirms that there are no genes that are absent in more than 75% of samples and so, for these filtering methods, we do not need to worry about faulty genes that are absent in all samples throwing off the experiment-by-experiment results or making all samples look more sparse than they are.


#### Experiments with too many 0's

Using the set of genes that intersects the array probes, we can check for samples with too many unexpressed (or undetected) genes to add biological value to our compendium in which case including them will only add noise.

The array compendium has a narrower distribution of sparse samples with the maximum number of undected genes (for biological or technical reasons) being less than 1,200 , or `r 120000 / nrow(comp_pao1_tpm)` %, of genes.

```{r, warning=F}
#array_zeros <- apply(comp_pao1_tpm[,c(1:ncol(array_comp))], 2, FUN = function(x) sum( x  == 0))
# to determine noise level since arrays have background noise
hist(unlist(array_comp))
array_zeros <- apply(array_comp, 2, FUN = function(x) sum( x  < 5))
hist(array_zeros, main = "Gene Expression < 5, by Sample", xlab = '# genes < 5')
q <- quantile(array_zeros, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
abline(v = q[10], col='red', lty=2)

```

```{r, warning=F}

#pao1_seq_zeros <- apply(comp_pao1_tpm[,c((ncol(array_comp)+1):ncol(comp_pao1_tpm))], 2, FUN = function(x) sum( x  == 0))
#pa14_seq_zeros <- apply(comp_pa14_tpm[,c((ncol(array_comp)+1):ncol(comp_pa14_tpm))], 2, FUN = function(x) sum( x  == 0))

pao1_seq_zeros <- apply(rnaseq_pao1_counts, 2, FUN = function(x) sum( x  == 0))
pa14_seq_zeros <- apply(rnaseq_pa14_counts, 2, FUN = function(x) sum( x  == 0))

par(mfrow=c(1,2))
hist(pao1_seq_zeros, main = 'PAO1')
q <- quantile(pao1_seq_zeros, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
#abline(v = q[1], col='red', lty=2)
abline(v = q[10], col='red', lty=2)

hist(pa14_seq_zeros, main = 'PA14')
q <- quantile(pa14_seq_zeros, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
#abline(v = q[1], col='red', lty=2)
abline(v = q[10], col='red', lty=2)


```
Using the above distributions we can determine the cutoff that will capture 95% of the samples.



## Experiments that are not sparse enough

In addition to samples being too sparse, RNAseq samples that are not sparse enough can be indicative of technical complications such as DNA contamination. This is identifiable when genes that are typically very lowly expressed are unexpetedly highly expressed in a given sample.

Knowing the wide range of gene expression values captured in RNAseq, it is adventageous to visualize counts and/or TPM values after log transformation. Note that zeros values remain zeros after transformation.

#### Trends of gene expression

Can sparsity be attributable to the expression patterns of a subset of genes

```{r, warning=F}
comp_pao1_tpm_log <- data.frame(log(data.matrix(comp_pao1_tpm)))
comp_pao1_tpm_log[comp_pao1_tpm_log == -Inf] <- 0
comp_pao1_tpm_log[comp_pao1_tpm_log < 0] <- 0

comp_pa14_tpm_log <- data.frame(log(data.matrix(comp_pa14_tpm)))
comp_pa14_tpm_log[comp_pa14_tpm_log == -Inf] <- 0
comp_pa14_tpm_log[comp_pa14_tpm_log < 0] <- 0
```

```{r}
c1 <- rgb(173,216,230, max = 255, alpha = 100, names = "lt.blue")
c2 <- rgb(255,192,203, max = 255, alpha = 50, names = "lt.pink")
c3 <- rgb(173,216,203, max = 255, alpha = 100, names = "lt.green")
c4 <- rgb(203,173,203, max = 255, alpha = 50, names = "lt.red")
```



To get an idea of how individual gene expression distributions across the RNAseq compendiia looks, I'll plot distributions for some randomly selected genes.
```{r, fig.height=6, fig.width=6}
par(mfrow=c(4,4))
for(i in c(1:8)){
  gene_r <- sample(intersect(rownames(comp_pao1_tpm_log),rownames(comp_pa14_tpm_log)),1)
  h1 <- hist(as.numeric(comp_pao1_tpm_log[gene_r, colnames(comp_pao1_tpm_log) %in% colnames(array_comp)]), plot = F, breaks=50)
  h2 <- hist(as.numeric(comp_pao1_tpm_log[gene_r, colnames(comp_pao1_tpm_log) %in% colnames(rnaseq_pao1_tpm)]), plot = F, breaks=50)
  plot(h1, col= c2,ylim=c(0,max(h1$counts, h2$counts)), xlim=c(min(h1$breaks, h2$breaks) , max(h1$breaks, h2$breaks))  , main = paste(gene_r,PAO1_to_name(gene_r)), xlab = "log(TPM)") # , 
  plot(h2, col= c1, add=T )
  
  h1 <- hist(as.numeric(comp_pa14_tpm_log[gene_r, colnames(comp_pa14_tpm_log) %in% colnames(array_comp)]), plot = F, breaks=50)
  h2 <- hist(as.numeric(comp_pa14_tpm_log[gene_r, colnames(comp_pa14_tpm_log) %in% colnames(rnaseq_pa14_tpm)]), plot = F, breaks=50)
  plot(h1, col= c2,ylim=c(0,max(h1$counts, h2$counts)), xlim=c(min(h1$breaks, h2$breaks) , max(h1$breaks, h2$breaks))  , main = paste(gene_r,PAO1_to_name(gene_r)), xlab = "log(TPM)") # , 
  plot(h2, col= c1, add=T )
}

```
It seems there are at least two types of distributions: generally highly expressed genes and generally lowly expressed or unexpressed genes. These two types of distributions may be distinguishable by their medians. If these are the two dominant distributions, and they have distinct medians, we may expect a bimodal distribution of per gene median gene expression across the compendium. 

Rather than plotting medians, the distributions may be distinguishable my their 25th percentile values.

```{r, warning=F}

par(mfrow=c(1,2))
medians <- apply(comp_pao1_tpm_log[,c(ncol(array_comp):ncol(comp_pao1_tpm_log))], 1, FUN = function(x) quantile(x, probs = c(.25,.75,.25,1))[3])
hist(medians, main = "First Quartile of Experiments  \n (per gene)", xlab = "Gene Expression")

medians <- apply(comp_pa14_tpm_log[,c(ncol(array_comp):ncol(comp_pa14_tpm_log))], 1, FUN = function(x) quantile(x, probs = c(.25,.75,.25,1))[3])
hist(medians, main = "First Quartile of Experiments  \n (per gene)", xlab = "Gene Expression")
```
A fair number of genes are lowly expressed at least 25% of the time. 

Ww can then expect the distributions of gene expression per experiment to contain a spike at 0, which makes the datasets sparse.

#### Experiment distributions

Similar to examining expression distributions gene by gene, we can look at each experiment, sample, datasets. 

```{r, warning=F, fig.height=6, fig.width=6}
pdf("experiment_dists.pdf", width=6,height=10)
par(mfrow=c(4,4))
for(i in c(1:8)){
  gene_r <- sample(colnames(comp_pao1_tpm_log),1)
  h1 <- hist(as.numeric(comp_pao1_tpm_log[rownames(comp_pao1_tpm_log), gene_r]), plot = F, breaks=50)
  h2 <- hist(as.numeric(comp_pao1_tpm_log[rownames(comp_pao1_tpm_log), gene_r]), plot = F, breaks=50)
  plot(h1, col= c2,ylim=c(0,max(h1$counts, h2$counts)), xlim=c(min(h1$breaks, h2$breaks) , max(h1$breaks, h2$breaks))  ,  main = paste(substr(gene_r,1,regexpr('\\.',gene_r)-1),'\n PAO1'),xlab = 'log(TPM)') # , 
  #plot(h2, col= c1, add=T )
  

  h1 <- hist(as.numeric(comp_pa14_tpm_log[rownames(comp_pa14_tpm_log), gene_r]), plot = F, breaks=50)
  h2 <- hist(as.numeric(comp_pa14_tpm_log[rownames(comp_pa14_tpm_log), gene_r]), plot = F, breaks=50)
  plot(h1, col= c2,ylim=c(0,max(h1$counts, h2$counts)), xlim=c(min(h1$breaks, h2$breaks) , max(h1$breaks, h2$breaks))  , main = paste(substr(gene_r,1,regexpr('\\.',gene_r)-1),'\n PA14'),xlab = 'log(TPM)') # , 
  #plot(h2, col= c1, add=T )
}
dev.off()
```
```{r}

par(mfrow=c(1,2))
medians <- apply(comp_pao1_tpm_log[,c(ncol(array_comp):ncol(comp_pao1_tpm_log))], 2, FUN = function(x) quantile(x, probs = c(.25,.75,.125,1))[3])
hist(medians, main = "First 12.5% of Genes PAO1  \n (per experiment)", xlab = "Gene Expression")

medians <- apply(comp_pa14_tpm_log[,c(ncol(array_comp):ncol(comp_pa14_tpm_log))], 2, FUN = function(x) quantile(x, probs = c(.25,.75,.125,1))[3])
hist(medians, main = "First 12.5% of Genes PA14  \n (per experiment)", xlab = "Gene Expression")
```
For most experiments at least 10% of the genes are lowly expressed.

```{r}
pdf("peaks_dists.pdf", width=6, height = 3)

par(mfrow=c(1,2))
medians <- apply(comp_pao1_tpm_log[,c(ncol(array_comp):ncol(comp_pao1_tpm_log))], 2, FUN = function(x) quantile(x, probs = c(.25,.75,.025,1))[3])
h1 <- hist(medians, breaks=50, main = "First 2.5% of Genes PAO1  \n (per experiment)", xlab = "Gene Expression")
#plot(h1)
abline(v = h1$breaks[2], col='red', lty=2)

medians <- apply(comp_pa14_tpm_log[,c(ncol(array_comp):ncol(comp_pa14_tpm_log))], 2, FUN = function(x) quantile(x, probs = c(.25,.75,.025,1))[3])
h1 <- hist(medians, breaks = 50, main = "First 2.5% of Genes PA14  \n (per experiment)", xlab = "Gene Expression")
#plot(h1)
abline(v = h1$breaks[2], col='red', lty=2)
dev.off()
```


### Load Annotations

Experiments in which nearly all genes have a detectable level of expression may be indicative of DNA contamination.

```{r}
run_table <- read.csv('SraRunTable.csv', stringsAsFactors = F)
arr_anns <- read.csv('~/Dropbox (Hogan Lab)/Resources/Annotations/Pa_Compendium_annotations.csv', stringsAsFactors = F)
```

First load the annotations to see how many samples there are in each experiment.
```{r, warning=F}
ann_exp_groups <- sapply(colnames(comp_pao1_tpm)[], function(x){
  exp <- if(x %in% arr_anns$CEL.file){
    arr_anns$Experiment[arr_anns$CEL.file == x]
  } else if(gsub('-','',gsub('\\.','',gsub('_','',x))) %in% gsub('-','',gsub('\\.','',gsub('_','',arr_anns$CEL.file)))){
    arr_anns$Experiment[gsub('-','',gsub('\\.','',gsub('_','',arr_anns$CEL.file))) == gsub('-','',gsub('\\.','',gsub('_','',x))) ]
  }  else if(substr(x,1,regexpr('\\.',x)[1]-1) %in% run_table$Experiment){
    run_table$SRA_study[ run_table$Experiment == substr(x,1,regexpr('\\.',x)[1]-1)]
  }   else{
    NA
  }
  exp[1]
  }
)

kable_classic(kable(head(sort(table(ann_exp_groups), decreasing = T), n=10), col.names = c("SRA Experiment","total # of samples"),
      caption = "10 largest experiments in the compendium and the number of samples each has"))
```

Now we can check if the unsparse samples come from the same experiments.
```{r}
un_sparse_exps <- sapply( c((ncol(array_comp)+1):ncol(comp_pao1_tpm_log)), function(i) { #)

  gene_r <- colnames(comp_pao1_tpm_log)[i]

  h1 <- hist(as.numeric(comp_pao1_tpm_log[,gene_r]), plot = F, breaks=50)
  
  #if(h1$counts[1] < max(h1$counts[-1])*0.15){
  
  if(h1$density[1] < .1275){
    ann_exp_groups[i]
  } else {""}
})

kable_classic(kable(head(sort(table(un_sparse_exps[un_sparse_exps != '']), decreasing = T), n=10), col.names = c("SRA Experiment","# of unsparse samples"),,
              caption = "Experiments that contain the most un-sparse samples"
              ))
```
We can visualize some of the un-sparse samples to confirm their distributions are missing the spike at 0.

```{r, fig.height=6, fig.width=6}
pdf("experiment_dists_nonsparse.pdf", width=6,height=10)
par(mfrow=c(4,4))
for(i in c(1:8)){
  gene_r <- sample(names(unsparse),1)
  h1 <- hist(as.numeric(comp_pao1_tpm_log[rownames(comp_pao1_tpm_log), gene_r]), plot = F, breaks=50)
  h2 <- hist(as.numeric(comp_pao1_tpm_log[rownames(comp_pao1_tpm_log), gene_r]), plot = F, breaks=50)
  plot(h1, col= c2,ylim=c(0,max(h1$counts, h2$counts)), xlim=c(min(h1$breaks, h2$breaks) , max(h1$breaks, h2$breaks))  , main = paste(substr(gene_r,1,regexpr('\\.',gene_r)-1),'\n PAO1'), xlab = 'log(TPM)') # , 
  #plot(h2, col= c1, add=T )
  

  h1 <- hist(as.numeric(comp_pa14_tpm_log[rownames(comp_pa14_tpm_log), gene_r]), plot = F, breaks=50)
  h2 <- hist(as.numeric(comp_pa14_tpm_log[rownames(comp_pa14_tpm_log), gene_r]), plot = F, breaks=50)
  plot(h1, col= c2,ylim=c(0,max(h1$counts, h2$counts)), xlim=c(min(h1$breaks, h2$breaks) , max(h1$breaks, h2$breaks))  , main = paste(substr(gene_r,1,regexpr('\\.',gene_r)-1),'\n PA14'), xlab = 'log(TPM)') # , 
  #plot(h2, col= c1, add=T )
}
dev.off()
```

These experiments lack the peak of genes with little to know expression and are more "dense" than the typical sparse dataset. If we take sparsity to be the true biological state of most cells, one that was not captured by array due to the compressed dynamic range, then datasets like this likely represent random or technical noise instead of useful biological information. For this possibility we are going to exclude them from this iteration of the compendium.

## 4. Genes by Expected patterns

Using genes known to be stable in their expression as a normalization metric is commonly used in qRT analysis and other mRNA quantification techniques. It can provide internal controls and account for technical biases. The same concept can hold true when we look at gene expression across compendia. We can choose housekeeping genes for **P. aeruginosa** based on a publication and examine their expression in the array compendium.

#### Experiments with unexpected expression in HK genes

The following set of HK genes comes from Alqarni et al 2016, J Microbiol Methods. The lowly expressed genes were determined based on expression in the array compendium (see other analyses).

```{r}
hks <- sapply(c('ppiD','rpoD','rpoS','proC','recA','rpsL','rho','oprL','tipA','nadB'), 
              function(x) name_to_PAO1(x))
hks <- hks[hks %in% rownames(comp_pao1_tpm_log)]

lgs <- c('PA2228','PA2227','PA2226','PA2225','PA2223','PA2222',                   
         'PA0442','PA0498','PA0715','PA0716','PA0984','PA0993')
lgs <- lgs[lgs %in% rownames(comp_pao1_tpm_log)]

hks_pa14 <- hks[hks %in% rownames(comp_pa14_tpm)]
lgs_pa14 <- lgs[lgs %in% rownames(comp_pa14_tpm)]
```


```{r}
pdf("hk_genes.pdf", width=6,height=3)
par(mfrow=c(1,2))
hk_means <- apply(comp_pao1_tpm_log[hks,], 2, FUN = function(x) mean(x))
hist(hk_means, main = 'PAO1-aligned')
q <- quantile(hk_means, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,1))
abline(v = q[1], col='red', lty=2)
abline(v = q[10], col='red', lty=2)

hk_means <- apply(comp_pa14_tpm_log[hks_pa14,], 2, FUN = function(x) mean(x))
hist(hk_means, main = 'PA14-aligned')
q <- quantile(hk_means, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,1))
abline(v = q[1], col='red', lty=2)
abline(v = q[10], col='red', lty=2)
dev.off()
```
Overall, the HK genes apepar to be relatively highly expressed and have a fairly narrow range thus allowing for outliers to be excluded based on a band-pass filter.


```{r, warning=F}
pdf("legs_dists.pdf", width=6,height=3)
par(mfrow=c(1,2))
lg_means <- apply(comp_pao1_tpm_log[lgs,], 2, FUN = function(x) mean(x))
hist(lg_means, main = 'PAO1-aligned')
q <- quantile(hk_means, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,1))
abline(v = q[1], col='red', lty=2)
#abline(v = q[10], col='red', lty=2)

lg_means <- apply(comp_pa14_tpm_log[lgs_pa14,], 2, FUN = function(x) mean(x))
hist(lg_means, main = 'PA14-aligned')
q <- quantile(hk_means, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,1))
abline(v = q[1], col='red', lty=2)
#abline(v = q[10], col='red', lty=2)
dev.off()
```
Especially in the PAO1-aligned compendium, the genes expected to be lowly expressed based on their expression in the array compendium are indeed lowly expressed, certainly in comparison to housekeeping genes. Although the distribution in less 0-skewed than the PAO1-aligned compendium, it still is lowly expressed compared to the housekeeping genes.

## 5. Medians 

In addition to looking at small sets of genes whose expression patterns are thought to be somewhat known, looking at the overall median of gene expression can be a proxy for technical variation. 
```{r}
pdf("medians_dists.pdf", width=6,height=3)
par(mfrow=c(1,2))

#unsp_exp <- sapply(colnames(data_comp_log), function(x) x %in% names(unsparse ))
medians <- apply(comp_pao1_tpm_log, 2, FUN = function(x) median(x))
hist(medians, main = 'PAO1-aligned')
q <- quantile(medians, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
abline(v = q[1], col='red', lty=2)
abline(v = q[10], col='red', lty=2)

#unsp_exp <- sapply(colnames(data_comp_log), function(x) x %in% names(unsparse ))
medians <- apply(comp_pa14_tpm_log, 2, FUN = function(x) median(x))
q <- quantile(medians, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
hist(medians, main = 'PA14-aligned')
abline(v = q[1], col='red', lty=2)
abline(v = q[10], col='red', lty=2)
dev.off()
```
These distributions have a clear shape that peaks around 3 logTPM with a small number of samples making up an outlier population of samples with very low medians.

However, this variability may not be indicative of technical failures and therefore maybe should not be a criterion for filtering but rather something to be corrected for. Many RNAseq pre-processing pipelines, including many scRNAseq ones, include a step of median-based normalization. For example, this is done as part of the EdgeR workflow I have used for differential expression analysis.

```{r}
rnaseq_pao1_count_mat <- ceiling(data.matrix(rnaseq_pao1_counts))
dds <- DESeqDataSetFromMatrix(countData = rnaseq_pao1_count_mat, colData = data.frame('exp'=ann_exp_groups[colnames(rnaseq_pao1_counts)]), design = ~ exp)
dds <- estimateSizeFactors(dds)
rnaseq_pao1_count_normalized_counts <- counts(dds, normalized=TRUE)
```


## 6. Set Final Criteria



```{r}

pao1_zeros <- quantile(apply(comp_pao1_tpm[,c((ncol(array_comp)+1):ncol(comp_pao1_tpm))][seq_strain_order == 'PAO1'], 2, FUN = function(x) sum( x  == 0)), probs = c(.7,.975))
pa14_zeros <- quantile(apply(comp_pa14_tpm[,c((ncol(array_comp)+1):ncol(comp_pao1_tpm))][seq_strain_order == 'PA14'], 2, FUN = function(x) sum( x  == 0)), probs = c(.7,.975))

#pao1_peak = 0.1 # this may need the most adjusting
#pa14_peak = 0.1

pao1_peak = 0.0 # this may need the most adjusting
pa14_peak = 0.0


pao1_hk <- quantile(apply(comp_pao1_tpm_log[hks,], 2, FUN = function(x) mean(x)), probs = c(.025,.975))
pa14_hk <- quantile(apply(comp_pa14_tpm_log[hks_pa14,], 2, FUN = function(x) mean(x)), probs = c(.025,.975))

#pao1_med <- quantile(apply(comp_pao1_tpm_log, 2, FUN = function(x) median(x)), probs = c(.05,.95))
#pa14_med <- quantile(apply(comp_pa14_tpm_log, 2, FUN = function(x) median(x)), probs = c(.05,.95))

pao1_med <- quantile(apply(comp_pao1_tpm_log, 2, FUN = function(x) median(x)), probs = c(.0,1))
pa14_med <- quantile(apply(comp_pa14_tpm_log, 2, FUN = function(x) median(x)), probs = c(.0,1))

```

```{r}
pdf('zeros_dists.pdf', width = 6, height=3)
par(mfrow=c(1,2))
hist(log(pao1_seq_zeros), main = 'PAO1')
#q <- quantile(pao1_seq_zeros, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
#abline(v = q[1], col='red', lty=2)
abline(v = log(pao1_zeros[1]), col='red', lty=2)
abline(v = log(pao1_zeros[2]), col='red', lty=2)

hist(log(pa14_seq_zeros), main = 'PA14')
#q <- quantile(pa14_seq_zeros, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
#abline(v = q[1], col='red', lty=2)
abline(v = log(pa14_zeros[1]), col='red', lty=2)
abline(v = log(pa14_zeros[2]), col='red', lty=2)

dev.off()
```

```{r}
pdf('zeros_dists_detail.pdf', width = 6, height=3)
par(mfrow=c(1,2))
hist(log(pao1_seq_zeros), main = 'PAO1', breaks=25, xlab = 'log(zeros)')
q <- quantile(pao1_seq_zeros, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
abline(v = log(q[1]), col='red', lty=2)
abline(v = log(q[2]), col='red', lty=2)
abline(v = log(q[3]), col='red', lty=2)
abline(v = log(q[4]), col='red', lty=2)
abline(v = log(q[5]), col='red', lty=2)
abline(v = log(q[10]), col='red', lty=2)
#abline(v = log(pao1_zeros[1]), col='red', lty=2)
#abline(v = log(pao1_zeros[2]), col='red', lty=2)

hist(log(pa14_seq_zeros), main = 'PA14',breaks=25, xlab = 'log(zeros)')
q <- quantile(pa14_seq_zeros, probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
abline(v = log(q[1]), col='red', lty=2)
abline(v = log(q[2]), col='red', lty=2)
abline(v = log(q[3]), col='red', lty=2)
abline(v = log(q[4]), col='red', lty=2)
abline(v = log(q[5]), col='red', lty=2)
abline(v = log(q[10]), col='red', lty=2)
#abline(v = log(pa14_zeros[1]), col='red', lty=2)
#abline(v = log(pa14_zeros[2]), col='red', lty=2)
dev.off()
```


```{r}
pdf('zeros_dists_strain_specific.pdf', width = 6, height=3)
par(mfrow=c(1,2))
hist(log(pao1_seq_zeros[seq_strain_order == 'PAO1']), main = 'PAO1', breaks=25, xlab = 'log(zeros)')
q <- quantile(pao1_seq_zeros[seq_strain_order == 'PAO1'], probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
abline(v = log(q[1]), col='red', lty=2)
abline(v = log(q[2]), col='red', lty=2)
abline(v = log(q[3]), col='red', lty=2)
abline(v = log(q[4]), col='red', lty=2)
abline(v = log(q[5]), col='red', lty=2)
abline(v = log(q[10]), col='red', lty=2)
#abline(v = log(pao1_zeros[1]), col='red', lty=2)
#abline(v = log(pao1_zeros[2]), col='red', lty=2)

hist(log(pa14_seq_zeros[seq_strain_order == 'PA14']), main = 'PA14',breaks=25, xlab = 'log(zeros)')
q <- quantile(pa14_seq_zeros[seq_strain_order == 'PA14'], probs = c(.1,.2,.3,.4,.5,.6,.7,.8,.9,.95,1))
abline(v = log(q[1]), col='red', lty=2)
abline(v = log(q[2]), col='red', lty=2)
abline(v = log(q[3]), col='red', lty=2)
abline(v = log(q[4]), col='red', lty=2)
abline(v = log(q[5]), col='red', lty=2)
abline(v = log(q[10]), col='red', lty=2)
#abline(v = log(pa14_zeros[1]), col='red', lty=2)
#abline(v = log(pa14_zeros[2]), col='red', lty=2)
dev.off()
```


# Summary Plots

#### Prepare data with log transformation
##### PAO1-aligned
```{r}
comp_pao1_tpm_log <- data.frame(log(data.matrix(comp_pao1_tpm)))
comp_pao1_tpm_log[comp_pao1_tpm_log == -Inf] <- 0
comp_pao1_tpm_log[comp_pao1_tpm_log < 0] <- 0

rnaseq_pao1_tpm_log <- data.frame(log(data.matrix(rnaseq_pao1_tpm)))
rnaseq_pao1_tpm_log[rnaseq_pao1_tpm_log == -Inf] <- 0
rnaseq_pao1_tpm_log[rnaseq_pao1_tpm_log < 0] <- 0
```

##### PA14-aligned

```{r}
comp_pa14_tpm_log <- data.frame(log(data.matrix(comp_pa14_tpm)))
comp_pa14_tpm_log[comp_pa14_tpm_log == -Inf] <- 0
comp_pa14_tpm_log[comp_pa14_tpm_log < 0] <- 0

rnaseq_pa14_tpm_log <- data.frame(log(data.matrix(rnaseq_pa14_tpm)))
rnaseq_pa14_tpm_log[rnaseq_pa14_tpm_log == -Inf] <- 0
rnaseq_pa14_tpm_log[rnaseq_pa14_tpm_log < 0] <- 0
```

#### Apply filters

##### PAO1-aligned
```{r}
#filt_sp_comp_pao1_tpm <- filter_sparsity(comp_pao1_tpm_log[rownames(comp_pao1_tpm),], max_zeros=2500, min_peak=0.1)
#filt_hk_comp_pao1_tpm <- filter_hks(comp_pao1_tpm_log[rownames(comp_pao1_tpm),], hk_min=4.5, hk_max=6.5)
#filt_md_comp_pao1_tpm <- apply(comp_pao1_tpm_log,2,FUN = function(x) median(x)) > 2 & apply(comp_pao1_tpm_log,2,FUN = function(x) median(x)) < 4.5

filt_sp_comp_pao1_tpm <- filter_sparsity(comp_pao1_tpm_log[rownames(comp_pao1_tpm),], max_zeros=pao1_zeros[2], min_zeros=pao1_zeros[1], min_peak=pao1_peak)

filt_hk_comp_pao1_tpm <- filter_hks(comp_pao1_tpm_log[rownames(comp_pao1_tpm),], hk_min=pao1_hk[1], hk_max=pao1_hk[2])

filt_md_comp_pao1_tpm <- apply(comp_pao1_tpm_log,2,FUN = function(x) median(x)) >= pao1_med[1] & apply(comp_pao1_tpm_log,2,FUN = function(x) median(x)) <= pao1_med[2]

out_comp_pao1_tpm <- comp_pao1_tpm[rownames(comp_pao1_tpm), filt_sp_comp_pao1_tpm & filt_hk_comp_pao1_tpm & filt_md_comp_pao1_tpm]
out_comp_pao1_counts <- comp_pao1_counts[rownames(comp_pao1_tpm), filt_sp_comp_pao1_tpm & filt_hk_comp_pao1_tpm & filt_md_comp_pao1_tpm]
```

##### PA14-aligned
```{r}
filt_sp_comp_pa14_tpm <- filter_sparsity(comp_pa14_tpm_log[rownames(comp_pa14_tpm),], max_zeros=pao1_zeros[2],min_zeros=pao1_zeros[1], min_peak=pao1_peak)

filt_hk_comp_pa14_tpm <- filter_hks(comp_pa14_tpm_log[rownames(comp_pa14_tpm),], hk_min=pa14_hk[1], hk_max=pa14_hk[2])

filt_md_comp_pa14_tpm <- apply(comp_pa14_tpm_log,2,FUN = function(x) median(x)) > pa14_med[1] & apply(comp_pa14_tpm_log,2,FUN = function(x) median(x)) < pa14_med[2]

out_comp_pa14_tpm <- comp_pa14_tpm[rownames(comp_pa14_tpm), filt_sp_comp_pa14_tpm & filt_hk_comp_pa14_tpm & filt_md_comp_pa14_tpm]
out_comp_pa14_counts <- comp_pa14_counts[rownames(comp_pa14_tpm), filt_sp_comp_pa14_tpm & filt_hk_comp_pa14_tpm & filt_md_comp_pa14_tpm]
```

#### Prepare data without quantile-normalization

##### PAO1-aligned
```{r}
comp_pao1_tpm_nn <- array_match(array_comp, rnaseq_pao1_tpm, a_ref=F, fsqn=F)
comp_pao1_counts_nn <- array_match(array_comp, rnaseq_pao1_counts, a_ref=F, fsqn=F)

comp_pao1_tpm_nn_log <- data.frame(log(data.matrix(comp_pao1_tpm_nn)))
comp_pao1_tpm_nn_log[comp_pao1_tpm_nn_log == -Inf] <- 0
comp_pao1_tpm_nn_log[comp_pao1_tpm_nn_log < 0] <- 0
```

##### PA14-aligned
```{r}
comp_pa14_tpm_nn <- array_match(array_comp, rnaseq_pa14_tpm, a_ref=F, fsqn=F)
comp_pa14_counts_nn <- array_match(array_comp, rnaseq_pa14_counts, a_ref=F, fsqn=F)

comp_pa14_tpm_nn_log <- data.frame(log(data.matrix(comp_pa14_tpm_nn)))
comp_pa14_tpm_nn_log[comp_pa14_tpm_nn_log == -Inf] <- 0
comp_pa14_tpm_nn_log[comp_pa14_tpm_nn_log < 0] <- 0
```

#### Plot PAO1-aligned
```{r}
source('filter_functions.R')
```

```{r, warning=F, message=F}
pdf("pao1_heamtap_zp2.pdf", height = 6, width = 12)
genes_50_r <- sample(rownames(comp_pao1_tpm_nn_log),50)
annotated_heatmap_filt(comp_pao1_tpm_nn_log[c(hks,lgs, genes_50_r),],
                  comp_pao1_counts,
                  filt_sp_comp_pao1_tpm,
                  filt_hk_comp_pao1_tpm,
                  filt_md_comp_pao1_tpm)
dev.off()
```



#### Plot PA14-aligned

```{r, warning=F, message=F}
pdf("pa14_heamtap_zp2.pdf", height = 6, width = 12)
genes_50_r <- sample(rownames(comp_pa14_tpm_nn_log),50)
annotated_heatmap_filt(comp_pa14_tpm_nn_log[c(hks_pa14,lgs_pa14, genes_50_r),],
                  comp_pa14_counts,
                  filt_sp_comp_pa14_tpm,
                  filt_hk_comp_pa14_tpm,
                  filt_md_comp_pa14_tpm)
dev.off()
```




# Summary of filtered datasets




```{r}
# must meet all PAO1-based criteria
pao1_t <- table(ann_exp_groups[!
                                 (filt_sp_comp_pao1_tpm 
                                  & filt_hk_comp_pao1_tpm 
                                  & filt_md_comp_pao1_tpm)])
# must meet all PA14-based criteria
pa14_t <- table(ann_exp_groups[!
                                 (filt_sp_comp_pa14_tpm 
                                  & filt_hk_comp_pa14_tpm 
                                  & filt_md_comp_pa14_tpm)])

# must meet all PAO1-based AND PA14-based criteria
or_t<- table(ann_exp_groups[!
                              (filt_sp_comp_pao1_tpm 
                               & filt_hk_comp_pao1_tpm 
                               & filt_md_comp_pao1_tpm 
                               & filt_sp_comp_pa14_tpm 
                               & filt_hk_comp_pa14_tpm 
                               & filt_md_comp_pa14_tpm)])

# must meet either PAO1-based or PA14-based criteria
and_t <- table(ann_exp_groups[((!
                                (filt_sp_comp_pao1_tpm 
                                 & filt_hk_comp_pao1_tpm 
                                 & filt_md_comp_pao1_tpm))
                              & 
                                (!
                                   (filt_sp_comp_pa14_tpm
                                    & filt_hk_comp_pa14_tpm
                                    & filt_md_comp_pa14_tpm)))])

all_t <- table(ann_exp_groups)

ugs <- unique(c(names(pao1_t), names(pa14_t)))

summ_t <- data.frame('Experiment' = ugs, 'failed_PAO1_crit' = as.numeric(pao1_t[ugs]), 'failed_PA14_crit' = as.numeric(pa14_t[ugs]), 'failed_either' = as.numeric(or_t[ugs]), 'failed_both' = as.numeric(and_t[ugs]), 'samples_in_exp' = as.numeric(all_t[ugs]))
```




```{r, warning=F}
kable_classic(kable(summ_t[head(order(summ_t[,4], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Experiments containing the 10 most samples that were removed from the final, filtered compendium"))
```

```{r}
save_kable(
  kable_classic(kable(summ_t[head(order(summ_t[,4], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Experiments containing the most samples that were removed from the final, filtered compendium")),
  'filter_summary_table_pz7.pdf',
  bs_theme = "default",
  self_contained = TRUE,
  extra_dependencies = NULL,
  latex_header_includes = NULL,
  keep_tex = FALSE,
  density = 300
)
```
## With Strain Annotations
```{r, warning=F}
accessory_compendia <- read.csv('compendia/PAO1_PA14_accessory_expression.csv', stringsAsFactors = F, sep='\t')
strain <- accessory_compendia$Strain.type_pao1
strain2 <- accessory_compendia$Strain.type_pa14
temp <- accessory_compendia$X


seq_strain_order <- unlist(sapply(colnames(rnaseq_pao1_counts), function(x){
  num <- substr(x,1,regexpr('.',x, fixed=T)[1]-1)
  accessory_compendia$Strain.type_pao1[accessory_compendia$X == num][1]
}))
seq_strain_order[is.na(seq_strain_order)] <- "No annotation"

array_strain_order <- unlist(sapply(colnames(array_comp), function(x){
  arr_anns$Strain[arr_anns$CEL.file == x][1]
}))
array_strain_order[is.na(array_strain_order)] <- "No annotation"

array_strain_pro <- unlist(sapply(array_strain_order, function(x){
  if(grepl('PAO1',x)){
    'PAO1'
  } else if(grepl('PA14',x)){
    'PA14'
  } else if(grepl('PAK',x)){
    "PAK"
  } else if(grepl('linical isolate',x) | grepl('C.I.',x)){
    "Clinical Isolate"
  } else{
    "No annotation"
  }
}))

array_ph <- rep("Array", length(array_strain_order))
all_strains <- c(array_ph, seq_strain_order)
all_strains[is.na(all_strains)] <- "No annotations"
``` 

```{r}
ann_exp_groups[is.na(ann_exp_groups)] <- "No group"
all_t_s <- data.frame('groups' = ann_exp_groups,
                          'strain' = all_strains)
rownames(all_t_s) <- make.unique(as.character(all_t_s$groups))

summ_t_strain <- data.frame('Experiment' = ugs,'Strain' = all_t_s[ugs,'strain'], 'failed_PAO1_crit' = as.numeric(pao1_t[ugs]), 'failed_PA14_crit' = as.numeric(pa14_t[ugs]), 'failed_either' = as.numeric(or_t[ugs]), 'failed_both' = as.numeric(and_t[ugs]), 'samples_in_exp' = as.numeric(all_t[ugs]))

kable_classic(kable(summ_t_strain[head(order(summ_t_strain[,4], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Experiments containing the most samples that were removed from the final, filtered compendium"))

```
```{r}
save_kable(
  kable_classic(kable(summ_t_strain[head(order(summ_t_strain[,6], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Experiments containing the most samples that were removed from the final, filtered compendium")),
  'filter_summary_table_strains_zp2.pdf',
  bs_theme = "default",
  self_contained = TRUE,
  extra_dependencies = NULL,
  latex_header_includes = NULL,
  keep_tex = FALSE,
  density = 300
)
write.csv(summ_t_strain[head(order(summ_t_strain[,6], decreasing = T), n=Inf),],
          row.names = F,
          'filter_summary_table_strains_zp2.csv')
```

## Are the samples that fail one strain and not the other annotated as the expected strain?

```{r}
# must meet all PAO1-based criteria
strains_pao1_t <- table(all_strains[(!
                                 (filt_sp_comp_pao1_tpm 
                                  & filt_hk_comp_pao1_tpm 
                                  & filt_md_comp_pao1_tpm))
                                 &
                                   (
                                 (filt_sp_comp_pa14_tpm 
                                  & filt_hk_comp_pa14_tpm 
                                  & filt_md_comp_pa14_tpm))])


# must meet all PA14-based criteria
strains_pa14_t <- table(all_strains[(
                                 (filt_sp_comp_pao1_tpm 
                                  & filt_hk_comp_pao1_tpm 
                                  & filt_md_comp_pao1_tpm))
                                 &
                                   (!
                                 (filt_sp_comp_pa14_tpm 
                                  & filt_hk_comp_pa14_tpm 
                                  & filt_md_comp_pa14_tpm))])


# must meet either PAO1-based or PA14-based criteria
strains_and_t <- table(all_strains[((!
                                (filt_sp_comp_pao1_tpm 
                                 & filt_hk_comp_pao1_tpm 
                                 & filt_md_comp_pao1_tpm))
                              & 
                                (!
                                   (filt_sp_comp_pa14_tpm
                                    & filt_hk_comp_pa14_tpm
                                    & filt_md_comp_pa14_tpm)))])

strains_all_t <- table(all_strains)

strains_ugs <- unique(c(names(strains_pao1_t), names(strains_pa14_t)))

strains_summ_t <- data.frame( 'Strain' = strains_ugs, 'failed_PAO1_only' = as.numeric(strains_pao1_t[strains_ugs]), 'failed_PA14_only' = as.numeric(strains_pa14_t[strains_ugs]), 'failed_both' = as.numeric(strains_and_t[strains_ugs]), 'samples_of_strain' = as.numeric(strains_all_t[strains_ugs]))

strains_summ_t[is.na(strains_summ_t)] <- 0


kable_classic(kable(strains_summ_t[head(order(strains_summ_t[,4], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Strains that failed strain-specific filtering steps"))
```



```{r}
save_kable(
  kable_classic(kable(strains_summ_t[head(order(strains_summ_t[,4], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Strains that failed strain-specific filtering steps")),
  'strain_based_summary_zp2.pdf',
  bs_theme = "default",
  self_contained = TRUE,
  extra_dependencies = NULL,
  latex_header_includes = NULL,
  keep_tex = FALSE,
  density = 300
)
```

## Breakdown by specific test
```{r}
pao1_t<- table(ann_exp_groups[!
                              (filt_sp_comp_pao1_tpm 
                               & filt_hk_comp_pao1_tpm 
                               & filt_md_comp_pao1_tpm)])
pa14_t <- table(ann_exp_groups[!
                              (filt_sp_comp_pa14_tpm 
                               & filt_hk_comp_pa14_tpm 
                               & filt_md_comp_pa14_tpm)])

pao1_sp_t <- table(ann_exp_groups[!
                                 (filt_sp_comp_pao1_tpm 
                                 )])

pao1_med_t <- table(ann_exp_groups[!
                                 ( filt_md_comp_pao1_tpm)])


pao1_hk_t <- table(ann_exp_groups[!
                                 (
                                  filt_hk_comp_pao1_tpm 
                                  )])

pa14_sp_t <- table(ann_exp_groups[!
                                 (filt_sp_comp_pa14_tpm 
                                 )])

pa14_med_t <- table(ann_exp_groups[!
                                 ( filt_md_comp_pa14_tpm)])


pa14_hk_t <- table(ann_exp_groups[!
                                 (
                                  filt_hk_comp_pa14_tpm 
                                  )])



pao1_summ_t <- data.frame('Experiment' = ugs,
                          'Strain' = all_t_s[ugs,'strain'], 
                          'pao1_any' = as.numeric(pao1_t[ugs]),
                          'pa14_any' =as.numeric(pa14_t[ugs]),  
                          'pao1_HK' =  as.numeric(pao1_hk_t[ugs]), 
                          'pao1_median' = as.numeric(pao1_med_t[ugs]), 
                          'pao1_sparsity' = as.numeric(pao1_sp_t[ugs]),
                          'pa14_HK' = as.numeric(pa14_hk_t[ugs]), 
                          'pa14_median' = as.numeric(pa14_med_t[ugs]), 
                          'pa14_sparsity' = as.numeric(pa14_sp_t[ugs]),
                          'samples_in_exp' = as.numeric(all_t[ugs]))

pao1_summ_t[is.na(pao1_summ_t)] <- 0

kable_classic(kable(pao1_summ_t[head(order(pao1_summ_t[,3], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Experiments containing the most samples that were removed from the final, filtered compendium (PAO1 aligned)"))
```

```{r}
save_kable(
kable_classic(kable(pao1_summ_t[head(order(pao1_summ_t[,3], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Experiments containing the most samples that were removed from the final, filtered compendium (PAO1 aligned)")),
  'filter_summary_table_breakout_zp2.pdf',
  bs_theme = "default",
  self_contained = TRUE,
  extra_dependencies = NULL,
  latex_header_includes = NULL,
  keep_tex = FALSE,
  density = 300
)
write.csv(pao1_summ_t[head(order(pao1_summ_t[,3], decreasing = T), n=Inf),],
          row.names = F,
          'filter_summary_table_breakout_zp2.csv')
```

## Strain-sepcific sparsity

```{r}
pao1_t_s<- table(all_strains[!
                              (filt_sp_comp_pao1_tpm 
                               & filt_hk_comp_pao1_tpm 
                               & filt_md_comp_pao1_tpm)])
pa14_t_s<- table(all_strains[(filt_sp_comp_pao1_tpm 
                               & filt_hk_comp_pao1_tpm 
                               & filt_md_comp_pao1_tpm)
                             & (!
                              (filt_sp_comp_pa14_tpm 
                               & filt_hk_comp_pa14_tpm 
                               & filt_md_comp_pa14_tpm))])

pao1_sp_s <- table(all_strains[!
                                 (filt_sp_comp_pao1_tpm 
                                 )])

pao1_med_s <- table(all_strains[!
                                 ( filt_md_comp_pao1_tpm)])


pao1_hk_s <- table(all_strains[!
                                 (
                                  filt_hk_comp_pao1_tpm 
                                  )])

pa14_sp_s <- table(all_strains[!
                                 (filt_sp_comp_pa14_tpm 
                                 )])

pa14_med_s <- table(all_strains[!
                                 ( filt_md_comp_pa14_tpm)])


pa14_hk_s <- table(all_strains[!
                                 (
                                  filt_hk_comp_pa14_tpm 
                                  )])





```



```{r}
pao1_summ_t_s <- data.frame('Strain' = strains_ugs, 
                            'pao1_any' = as.numeric(pao1_t_s[strains_ugs]),
                            'pa14_any_only' =as.numeric(pa14_t_s[strains_ugs]),  
                            'pao1_HK' =  as.numeric(pao1_hk_s[strains_ugs]), 
                            'pao1_median' = as.numeric(pao1_med_s[strains_ugs]), 
                            'pao1_sparsity' = as.numeric(pao1_sp_s[strains_ugs]),
                            #'pa14_HK' = as.numeric(pa14_hk_s[strains_ugs]), 
                            #'pa14_median' = as.numeric(pa14_med_s[strains_ugs]), 
                            #'pa14_sparsity' = as.numeric(pa14_sp_s[strains_ugs]),
                            'samples_in_exp' = as.numeric(strains_all_t[strains_ugs]))

pao1_summ_t_s[is.na(pao1_summ_t_s)] <- 0

kable_classic(kable(pao1_summ_t_s[head(order(pao1_summ_t_s[,2], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Strains containing most samples that were removed from the final, filtered compendium (PAO1 aligned)"))
```
```{r}
save_kable(
kable_classic(kable(pao1_summ_t_s[head(order(pao1_summ_t_s[,2], decreasing = T), n=Inf),], 
                    row.names = F,
                    caption = "Strains with the most samples that were removed from the final, filtered compendium (PAO1 aligned)")),
  'filter_summary_table_strains_breakout_zp2.pdf',
  bs_theme = "default",
  self_contained = TRUE,
  extra_dependencies = NULL,
  latex_header_includes = NULL,
  keep_tex = FALSE,
  density = 300
)
write.csv(summ_t_strain[head(order(summ_t_strain[,2], decreasing = T), n=Inf),],
          row.names = F,
          'filter_summary_table_strains_breakout_zp2.csv')
```


## Spot checks of strain failures

What I really want are the the numbers of samples that failed based on one strain and not the other.

Fort he experiments with failed samples that I would liek to follow-up on, what is the experiment-wise breakdown?

```{r}
kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP288730"]),]))
kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP214490"]),]))
kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP188684"]),]))

kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP214490" & as.numeric(rowSums(filter_table_pao1)) >2]) ,]))
kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP214490" & as.numeric(rowSums(filter_table_pao1)) < 3]) ,]))


kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP173226" & as.numeric(rowSums(filter_table_pao1)) >2]) ,]))
kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP173226" & as.numeric(rowSums(filter_table_pao1)) < 3]) ,]))


kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP287307" & as.numeric(rowSums(filter_table_pao1)) >2]) ,]))
kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP287307" & as.numeric(rowSums(filter_table_pao1)) < 3]) ,]))


kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP037771" & as.numeric(rowSums(filter_table_pao1)) >2]) ,]))
kable_classic(kable(filter_table_pao1[names(ann_exp_groups[ann_exp_groups == "SRP037771" & as.numeric(rowSums(filter_table_pao1)) < 3]) ,]))
```

The things that catch my eye so far are:

1. We should allow samples to pass either PAO1 or PA14 checks but not require them to pass both. This makes sense as any given sample can only really be one strain.
SRP214490 is an experiment of 191 clinical isolate samples and 116 fail either the PAO1 or PA14 checks but only 47 fail both. An alternate approach is to only use "core" genes for the QC step though we will undoubtedly still have clinical isolates that lack certain genes.

2. The median-based filter is pulling out a couple large experiments whereas sparsity is pulling out a smattering of samples from many experiments.
I think the small number of large experiments we are loosing are of interest and would be a loss to exclude. I wonder if median is picking out batch affect but not necessarily low-quality such that correcting via median-based normalization will be true to the biology. 

This is testable looking at DNA contamination via reads of intergenic regions.

3. PA14-alignment based criteria seem to be filtering out more samples.
Many of the samples being filtered by PA14-alignment based criteria are annotated as PA14 even though some end up passing the PAO1-alignment based criteria only. This is mostly happening based on the PA14-driven sparsity check.


## 7. Write out Data

#### Prepare for filtering with log transformation

##### PAO1-aligned

```{r}
comp_pao1_tpm_log <- data.frame(log(data.matrix(comp_pao1_tpm)))
comp_pao1_tpm_log[comp_pao1_tpm_log == -Inf] <- 0
comp_pao1_tpm_log[comp_pao1_tpm_log < 0] <- 0

rnaseq_pao1_tpm_log <- data.frame(log(data.matrix(rnaseq_pao1_tpm)))
rnaseq_pao1_tpm_log[rnaseq_pao1_tpm_log == -Inf] <- 0
rnaseq_pao1_tpm_log[rnaseq_pao1_tpm_log < 0] <- 0
```


##### PA14-aligned

```{r}
comp_pa14_tpm_log <- data.frame(log(data.matrix(comp_pa14_tpm)))
comp_pa14_tpm_log[comp_pa14_tpm_log == -Inf] <- 0
comp_pa14_tpm_log[comp_pa14_tpm_log < 0] <- 0

rnaseq_pa14_tpm_log <- data.frame(log(data.matrix(rnaseq_pa14_tpm)))
rnaseq_pa14_tpm_log[rnaseq_pa14_tpm_log == -Inf] <- 0
rnaseq_pa14_tpm_log[rnaseq_pa14_tpm_log < 0] <- 0
```

#### Filter on criteria

##### PAO1-aligned



```{r}
filt_sp_comp_pao1_tpm <- filter_sparsity(comp_pao1_tpm_log[rownames(comp_pao1_tpm),], max_zeros=pao1_zeros[2], min_zeros=pao1_zeros[1], min_peak=pao1_peak)

filt_hk_comp_pao1_tpm <- filter_hks(comp_pao1_tpm_log[rownames(comp_pao1_tpm),], hk_min=pao1_hk[1], hk_max=pao1_hk[2])

filt_md_comp_pao1_tpm <- apply(comp_pao1_tpm_log,2,FUN = function(x) median(x)) >= pao1_med[1] & apply(comp_pao1_tpm_log,2,FUN = function(x) median(x)) <= pao1_med[2]
```

##### PA14-aligned

```{r}
filt_sp_comp_pa14_tpm <- filter_sparsity(comp_pa14_tpm_log[rownames(comp_pa14_tpm),], max_zeros=pao1_zeros[2],min_zeros=pao1_zeros[1], min_peak=pao1_peak)

filt_hk_comp_pa14_tpm <- filter_hks(comp_pa14_tpm_log[rownames(comp_pa14_tpm),], hk_min=pa14_hk[1], hk_max=pa14_hk[2])

filt_md_comp_pa14_tpm <- apply(comp_pa14_tpm_log,2,FUN = function(x) median(x)) > pa14_med[1] & apply(comp_pa14_tpm_log,2,FUN = function(x) median(x)) < pa14_med[2]
```

##### Joined Inclusive

```{r}
filt_sp_comp_jor_tpm <- filt_sp_comp_pao1_tpm | filt_sp_comp_pa14_tpm
filt_hk_comp_jor_tpm <- filt_hk_comp_pao1_tpm | filt_hk_comp_pa14_tpm
filt_md_comp_jor_tpm <- filt_md_comp_pao1_tpm | filt_md_comp_pa14_tpm
```

#### Write out csv files

```{r}
filter_table_pao1 <- data.frame("sparsity_test" = filt_sp_comp_pao1_tpm,
                                "housekeeping_genes_test" = filt_hk_comp_pao1_tpm,
                                "medians_test" = filt_md_comp_pao1_tpm)

filter_table_pa14 <- data.frame("sparsity_test" = filt_sp_comp_pa14_tpm,
                                "housekeeping_genes_test" = filt_hk_comp_pa14_tpm,
                                "medians_test" = filt_md_comp_pa14_tpm)

write.csv(filter_table_pao1,'pao1_aligned_compendium_filter_results_zp7.csv')
write.csv(filter_table_pa14,'pa14_aligned_compendium_filter_results_zp7.csv')
```


##### PAO1-aligned

```{r}
out_comp_pao1_tpm <- comp_pao1_tpm[rownames(comp_pao1_tpm), filt_sp_comp_pao1_tpm & filt_hk_comp_pao1_tpm & filt_md_comp_pao1_tpm]
out_comp_pao1_counts <- comp_pao1_counts[rownames(comp_pao1_tpm), filt_sp_comp_pao1_tpm & filt_hk_comp_pao1_tpm & filt_md_comp_pao1_tpm]

write.csv(out_comp_pao1_tpm, 'pao1_aligned_full_compendium_filtered_tpm_zp7.csv')
write.csv(out_comp_pao1_tpm, 'pao1_aligned_full_compendium_filtered_no_gene_names_tpm_zp7.csv', row.names = F)

write.csv(out_comp_pao1_counts, 'pao1_aligned_full_compendium_filtered_counts_zp7.csv')
write.csv(out_comp_pao1_counts, 'pao1_aligned_full_compendium_filtered_no_gene_names_counts_zp7.csv', row.names = F)

seq_out_comp_pao1_tpm <- rnaseq_pao1_tpm[,colnames(rnaseq_pao1_tpm) %in% colnames(out_comp_pao1_tpm)]
seq_out_comp_pao1_counts <- rnaseq_pao1_counts[,colnames(rnaseq_pao1_counts) %in% colnames(out_comp_pao1_counts)]

write.csv(seq_out_comp_pao1_tpm, 'pao1_aligned_rnaseq_compendium_filtered_tpm_zp7.csv')
write.csv(seq_out_comp_pao1_tpm, 'pao1_aligned_rnaseq_compendium_filtered_no_gene_names_tpm_zp7.csv', row.names = F)

write.csv(seq_out_comp_pao1_counts, 'pao1_aligned_rnaseq_compendium_filtered_counts_zp7.csv')
write.csv(seq_out_comp_pao1_counts, 'pao1_aligned_rnaseq_compendium_filtered_no_gene_names_counts_zp7.csv', row.names = F)
```

##### PA14-aligned

```{r}
out_comp_pa14_tpm <- comp_pa14_tpm[rownames(comp_pa14_tpm), filt_sp_comp_pa14_tpm & filt_hk_comp_pa14_tpm & filt_md_comp_pa14_tpm]
out_comp_pa14_counts <- comp_pa14_counts[rownames(comp_pa14_tpm), filt_sp_comp_pa14_tpm & filt_hk_comp_pa14_tpm & filt_md_comp_pa14_tpm]

rownames(out_comp_pa14_tpm) <- sapply(rownames(out_comp_pa14_tpm),
                                          function(x) PAO1_to_PA14(x))
rownames(out_comp_pa14_counts) <- sapply(rownames(out_comp_pa14_counts),
                                          function(x) PAO1_to_PA14(x))

write.csv(out_comp_pa14_tpm, 'pa14_aligned_full_compendium_filtered_tpm_zp7.csv')
write.csv(out_comp_pa14_tpm, 'pa14_aligned_full_compendium_filtered_no_gene_names_tpm_zp7.csv', row.names = F)

write.csv(out_comp_pa14_counts, 'pa14_aligned_full_compendium_filtered_counts_zp7.csv')
write.csv(out_comp_pa14_counts, 'pa14_aligned_full_compendium_filtered_no_gene_names_counts_zp7.csv', row.names = F)

seq_out_comp_pa14_tpm <- rnaseq_pa14_tpm[,colnames(rnaseq_pa14_tpm) %in% colnames(out_comp_pa14_tpm)]
seq_out_comp_pa14_counts <- rnaseq_pa14_counts[,colnames(rnaseq_pa14_counts) %in% colnames(out_comp_pa14_counts)]
rownames(seq_out_comp_pa14_tpm) <- sapply(rownames(seq_out_comp_pa14_tpm),
                                          function(x) PAO1_to_PA14(x))
rownames(seq_out_comp_pa14_counts) <- sapply(rownames(seq_out_comp_pa14_counts),
                                          function(x) PAO1_to_PA14(x))

write.csv(seq_out_comp_pa14_tpm, 'pa14_aligned_rnaseq_compendium_filtered_tpm_zp7.csv')
write.csv(seq_out_comp_pa14_tpm, 'pa14_aligned_rnaseq_compendium_filtered_no_gene_names_tpm_zp7.csv', row.names = F)

write.csv(seq_out_comp_pa14_counts, 'pa14_aligned_rnaseq_compendium_filtered_counts_zp7.csv')
write.csv(seq_out_comp_pa14_counts, 'pa14_aligned_rnaseq_compendium_filtered_no_gene_names_counts_zp7.csv', row.names = F)
```


##### Joined-inclusive-PAO1-aligned


```{r}
out_comp_jor_pao1_tpm <- comp_pao1_tpm[rownames(comp_pao1_tpm), filt_sp_comp_jor_tpm & filt_hk_comp_jor_tpm & filt_md_comp_jor_tpm]
out_comp_jor_pao1_counts <- comp_pao1_counts[rownames(comp_pao1_tpm), filt_sp_comp_jor_tpm & filt_hk_comp_jor_tpm & filt_md_comp_jor_tpm]
                                

write.csv(out_comp_jor_pao1_tpm, 'jor_pao1_aligned_full_compendium_filtered_tpm_zp7.csv')
write.csv(out_comp_jor_pao1_tpm, 'jor_pao1_aligned_full_compendium_filtered_no_gene_names_tpm_zp7.csv', row.names = F)

write.csv(out_comp_jor_pao1_counts, 'jor_pao1_aligned_full_compendium_filtered_counts_zp7.csv')
write.csv(out_comp_jor_pao1_counts, 'jor_pao1_aligned_full_compendium_filtered_no_gene_names_counts_zp7.csv', row.names = F)

seq_out_comp_jor_pao1_tpm <- rnaseq_pao1_tpm[,colnames(rnaseq_pao1_tpm) %in% colnames(out_comp_jor_pao1_tpm)]
seq_out_comp_jor_pao1_counts <- rnaseq_pao1_counts[,colnames(rnaseq_pao1_counts) %in% colnames(out_comp_jor_pao1_counts)]


write.csv(seq_out_comp_jor_pao1_tpm, 'jor_pao1_aligned_rnaseq_compendium_filtered_tpm_zp7.csv')
write.csv(seq_out_comp_jor_pao1_tpm, 'jor_pao1_aligned_rnaseq_compendium_filtered_no_gene_names_tpm_zp7.csv', row.names = F)

write.csv(seq_out_comp_jor_pao1_counts, 'jor_pao1_aligned_rnaseq_compendium_filtered_counts_zp7.csv')
write.csv(seq_out_comp_jor_pao1_counts, 'jor_pao1_aligned_rnaseq_compendium_filtered_no_gene_names_counts_zp7.csv', row.names = F)
2
```





##### Joined-inclusive-PA14-aligned

```{r}
out_comp_jor_pa14_tpm <- comp_pa14_tpm[rownames(comp_pa14_tpm), filt_sp_comp_jor_tpm & filt_hk_comp_jor_tpm & filt_md_comp_jor_tpm]
out_comp_jor_pa14_counts <- comp_pa14_counts[rownames(comp_pa14_tpm), filt_sp_comp_jor_tpm & filt_hk_comp_jor_tpm & filt_md_comp_jor_tpm]

rownames(out_comp_jor_pa14_tpm) <- sapply(rownames(out_comp_jor_pa14_tpm),
                                          function(x) PAO1_to_PA14(x))
rownames(out_comp_jor_pa14_counts) <- sapply(rownames(out_comp_jor_pa14_counts),
                                          function(x) PAO1_to_PA14(x))

write.csv(out_comp_jor_pa14_tpm, 'jor_pa14_aligned_full_compendium_filtered_tpm_zp7.csv')
write.csv(out_comp_jor_pa14_tpm, 'jor_pa14_aligned_full_compendium_filtered_no_gene_names_tpm_zp7.csv', row.names = F)

write.csv(out_comp_jor_pa14_counts, 'jor_pa14_aligned_full_compendium_filtered_counts_zp7.csv')
write.csv(out_comp_jor_pa14_counts, 'jor_pa14_aligned_full_compendium_filtered_no_gene_names_counts_zp7.csv', row.names = F)

seq_out_comp_jor_pa14_tpm <- rnaseq_pa14_tpm[,colnames(rnaseq_pa14_tpm) %in% colnames(out_comp_jor_pa14_tpm)]
seq_out_comp_jor_pa14_counts <- rnaseq_pa14_counts[,colnames(rnaseq_pa14_counts) %in% colnames(out_comp_jor_pa14_counts)]
rownames(seq_out_comp_jor_pa14_tpm) <- sapply(rownames(seq_out_comp_jor_pa14_tpm),
                                          function(x) PAO1_to_PA14(x))
rownames(seq_out_comp_jor_pa14_counts) <- sapply(rownames(seq_out_comp_jor_pa14_counts),
                                          function(x) PAO1_to_PA14(x))

write.csv(seq_out_comp_jor_pa14_tpm, 'jor_pa14_aligned_rnaseq_compendium_filtered_tpm_zp7.csv')
write.csv(seq_out_comp_jor_pa14_tpm, 'jor_pa14_aligned_rnaseq_compendium_filtered_no_gene_names_tpm_zp7.csv', row.names = F)

write.csv(seq_out_comp_jor_pa14_counts, 'jor_pa14_aligned_rnaseq_compendium_filtered_counts_zp7.csv')
write.csv(seq_out_comp_jor_pa14_counts, 'jor_pa14_aligned_rnaseq_compendium_filtered_no_gene_names_counts_zp7.csv', row.names = F)
```

## Visualizing Non-sparse samples

```{r}
unsparse_comp_pao1_tpm <- comp_pao1_tpm[rownames(comp_pao1_tpm), ! filt_sp_comp_pao1_tpm ]
```

```{r, warning=F} 
#, fig.height=6, fig.width=6
png("SRP161626_hists.png", width=10, height= 20, unit = "in", res=300)
par(mfrow=c(10,4))
for(i in c(1:40)){
  gene_r <- sample(colnames(comp_pao1_tpm_log)[ann_exp_groups == 'SRP161626'],1)
  h1 <- hist(as.numeric(comp_pao1_tpm_log[rownames(comp_pao1_tpm_log), gene_r]), plot = F, breaks=50, freq = F)
  h2 <- hist(as.numeric(comp_pao1_tpm_log[lgs, gene_r]), plot = F, breaks=25, freq = F)
  h3 <- hist(as.numeric(comp_pao1_tpm_log[hks, gene_r]), plot = F, breaks=25, freq = F)
  plot(h1, col= c2,ylim=c(0,max(h1$density, h2$density, h3$density)), xlim=c(min(h1$breaks, h2$breaks, h3$breaks) , max(h1$breaks, h2$breaks, h3$breaks))  ,  main = paste(paste(substr(gene_r,1,regexpr('\\.',gene_r)-1),'\n PAO1'), filt_sp_comp_pao1_tpm[gene_r]) ,xlab = 'log(TPM)', freq = F) # , 
  #plot(h2, col= c1, add=T )
  plot(h2, col= "blue", add=T , freq = F)
  plot(h3, col= "red", add=T , freq = F)
  

  #h1 <- hist(as.numeric(comp_pa14_tpm_log[rownames(comp_pa14_tpm_log), gene_r]), plot = F, breaks=50)
  #h2 <- hist(as.numeric(comp_pa14_tpm_log[lgs, gene_r]), plot = F, breaks=50)
  #h3 <- hist(as.numeric(comp_pa14_tpm_log[hks, gene_r]), plot = F, breaks=50)
  #plot(h1, col= c2,ylim=c(0,max(h1$counts, h2$counts)), xlim=c(min(h1$breaks, h2$breaks) , max(h1$breaks, h2$breaks))  , main = paste(substr(gene_r,1,regexpr('\\.',gene_r)-1),'\n PA14'),xlab = 'log(TPM)') # , 
  #plot(h2, col= c1, add=T )
  #plot(h3, col= c3, add=T )
}
dev.off()
```